docs/src/pages/guides/testing.md

```diff
@@ -0,0 +1,137 @@
+# Testing
+
+## Internal
+
+We take tests seriously, we have written and we maintain **a wide range** of tests so we can
+iterate with confidence on the components. For instance, the visual regression tests provided by [Argos-CI](https://www.argos-ci.com/callemall/material-ui) have proven to be really helpful.
+To learn more about our internal tests, you can have a look at the [README](https://github.com/callemall/material-ui/blob/next/test/README.md).
+
+[![Coverage Status](https://img.shields.io/codecov/c/github/callemall/material-ui/next.svg)](https://codecov.io/gh/callemall/material-ui/branch/next)
+
+## Userspace
+
+But what about writing tests in userspace? Material-UI styling infrastructure relies on the context feature of React to work correcly. You are going to need that context when testing your react components with ours.
+
+### Shallow rendering
+
+Shallow rendering is useful to constrain yourself to testing a component as a unit, and to ensure that your tests aren't indirectly asserting on behavior of child components.
+We expose a `createShallow()` function for that situation. However, you will most likely not need it for most of the simple situations.
+
+### Full DOM rendering
+
+Full DOM rendering is ideal for use cases where you have components that may interact with DOM APIs, or may require the full lifecycle in order to fully test the component (i.e., `componentDidMount` etc.).
+We expose a `createMount()` function for that situation.
+
+### Render to string
+
+Rendering to string is useful to test the behavior of the components that are used on the server.
+You can take advantage of it to assert the generated HTML string.
+We expose a `createRender()` function for that sitation.
+
+## API
+
+### `createShallow([options]) => shallow`
+
+Generate an enhanced shallow function with the needed context.
+You can find the API of the `shallow` function following [that link](http://airbnb.io/enzyme/docs/api/shallow.html).
```

Comment:
'this link' (although good practice is that the anchor text is somewhat descriptive of the target: 'Please refer to the [API documentation](http://airbnb.io/enzyme/docs/api/shallow.html) for further details of the `shallow function`.'